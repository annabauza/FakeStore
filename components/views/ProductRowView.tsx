import React from 'react';
import { View, Text, Image, Button, StyleSheet } from 'react-native';
import { NavigationScreenProp } from 'react-navigation';
import { Product } from '../../network/API';

export interface ProductRowViewProps {
  navigation?: NavigationScreenProp<any, any>;
  product: Product;
}

const ProductRowView: React.FC<ProductRowViewProps> = ({
  product: { id, title, image, price },
  navigation,
}) => {
  return (
    <View style={styles.containerColumn}>
      <View style={styles.containerRow}>
        <Image
          style={styles.thumbnail}
          source={{
            uri: image,
          }}
        />
        <Text style={styles.input}>Â£{price}</Text>
      </View>
      <Text style={styles.input}>{title}</Text>
      {navigation ? (
        <Button
          title="more >"
          onPress={() => {
            navigation?.navigate('ProductDetails', { productId: id });
          }}
        />
      ) : null}
      <View style={styles.divider} />
    </View>
  );
};

const styles = StyleSheet.create({
  containerColumn: {
    flex: 1,
    padding: 16,
    flexDirection: 'column',
  },
  containerRow: {
    flex: 1,
    width: '100%',
    padding: 16,
    flexDirection: 'row',
  },
  divider: {
    flex: 1,
    height: 1,
    backgroundColor: '#000',
    marginBottom: 12,
  },
  thumbnail: {
    flex: 1,
    width: 66,
    height: 66,
    resizeMode: 'contain',
  },
  input: {
    margin: 16,
  },
});

export default ProductRowView;
