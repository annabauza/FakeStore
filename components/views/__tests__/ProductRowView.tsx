import React from 'react';
import renderer from 'react-test-renderer';
import ProductRowView from '../ProductRowView';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <ProductRowView
        product={{
          id: 0,
          title: 'test',
          image: 'test.png',
          price: 11,
          category: 'test',
        }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
