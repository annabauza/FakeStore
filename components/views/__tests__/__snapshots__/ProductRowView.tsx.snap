// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "padding": 16,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "padding": 16,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "test.png",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 66,
          "resizeMode": "contain",
          "width": 66,
        }
      }
    />
    <Text
      style={
        Object {
          "margin": 16,
        }
      }
    >
      Â£
      11
    </Text>
  </View>
  <Text
    style={
      Object {
        "margin": 16,
      }
    }
  >
    test
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "#000",
        "flex": 1,
        "height": 1,
        "marginBottom": 12,
      }
    }
  />
</View>
`;
